### ISSUE TITLE
Hook up with the Downgrade Workspace API

### ISSUE THREAD

**Issue by carlosmiceli** [Expensify/App#51515]
Part of [Add Workspace Plan Downgrades to NewDot](https://github.com/Expensify/Expensify/issues/399936) 

[Doc section](https://docs.google.com/document/d/1GDPmbvV47VVxucr-tHYjH80VylcJptP0RfGMVy3zanY/edit?tab=t.0#bookmark=id.lc1fx7nifcvp)

Steps to complete:
- Create a new action
- Connect it with the downgrade button

**Comment by shawnborton** [comment]
I think we're still holding on this one, right?

**Comment by carlosmiceli** [comment]
Will try to make progress today.

**Comment by carlosmiceli** [comment]
We are close to being able to do this, final BE PR is in review and we can start testing FE and BE together.

**Comment by carlosmiceli** [comment]
Hey @DylanDylann, this should be ready to tackle now that the BE is complete :)

**Comment by sonialiap** [comment]
@DylanDylann this feature is ready for the FE part! Will you be able to handle this one?

**Comment by DylanDylann** [comment]
@carlosmiceli The API is up on my side. But I see a problem, Policy_DowngradeToTeam API returns lots of data and I see that there are two POLICYCHANGELOG_TEAM_DOWNGRADE report actions returned from BE

In my below video

Admin reportID: `8798864160178330` 

In this admin report, we have two POLICYCHANGELOG_TEAM_DOWNGRADE report actions with report action IDs: 1732067875874417562 and 2114614043739997111  

https://github.com/user-attachments/assets/52c77fd3-43b0-43d3-b008-233152882fe5

**Comment by carlosmiceli** [comment]
@DylanDylann still haven't been able to figure out why we're seeing double report actions, but can we move forward with the PR and hardcode using the correct one while I discover what could be the issue and make the BE fix?

**Comment by carlosmiceli** [comment]
@youssef-lr do you maybe have any suggestion about what could be happening here? I'm looking at the logs and I still haven't figured out yet where and why we are creating two report actions, but maybe you have a suggestions of where to look? [Here](https://github.com/Expensify/Auth/blob/5c16462b37d2f6c61c4484b4b8f28aa5bcf37c59/auth/command/SavePolicy.cpp#L549) is where we add the action at the moment.

**Comment by DylanDylann** [comment]
@carlosmiceli Yeah I already did the FE implementation locally with a branch built up from https://github.com/Expensify/App/pull/53343. I will raise PR after https://github.com/Expensify/App/pull/53343 is merged

**Comment by carlosmiceli** [comment]
Great!

**Comment by sonialiap** [comment]
I'm OOO Dec 16-20, I don't think this will need a BZ before I'm back, but if it does please reapply the bug label

**Comment by carlosmiceli** [comment]
Not overdue, we're working on it :D

**Comment by carlosmiceli** [comment]
@DylanDylann I think I found the issue with the duplicate action, but can we proceed using the first one while I try to fix it?

**Comment by carlosmiceli** [comment]
Fixed, will be merged soon. The action that says `this workspace` instead of the policy name is the right one.

**Comment by DylanDylann** [comment]
Please ping me when the change is deployed

**Comment by carlosmiceli** [comment]
It's now deployed.

**Comment by DylanDylann** [comment]
@carlosmiceli Worked well

<img width="878" alt="Screenshot 2024-12-19 at 11 39 05" src="https://github.com/user-attachments/assets/da3c029a-d5f0-4a91-8103-7e8f261342eb" />

**Comment by carlosmiceli** [comment]
Fantastic, thanks sir! cc @garrettmknight 

I'll go over all the open issues tomorrow to see what's left to do, but anything outstanding from your end @DylanDylann ?

**Comment by DylanDylann** [comment]
Everything works well from me

**Comment by carlosmiceli** [comment]
To confirm, [this](https://github.com/Expensify/App/issues/51515#issuecomment-2540595296) is still the plan, correct?

**Comment by sonialiap** [comment]
Looks like the deploy blocker was due to a BE change so not counting as a regression https://github.com/Expensify/App/issues/54499#issuecomment-2561444017

**Comment by sonialiap** [comment]
Payment summary:
- @DylanDylann $250 - offer sent in [upwork](https://www.upwork.com/nx/wm/offer/105546663)

**Comment by DylanDylann** [comment]
@sonialiap Sorry, but could you make a refund here? The payment will be handled later

cc @carlosmiceli

**Comment by DylanDylann** [comment]
@sonialiap I already made a refund on Upwork because this issue belongs to a project and we will handle all payments in one later

cc @carlosmiceli
